content.js

```diff
@@ -2,7 +2,139 @@ const ISSUE_URL_REGEX = (
   /^https:\/\/sentry.prod.mozaws.net\/operations\/nightly-js-errors\/issues\/([0-9]+)\/(?:events\/([0-9]+)\/)?/
 );
 
+const sentry = {
+  issueCache: new Map(),
+  eventCache: new Map(),
+
+  async _get(url) {
+    try {
+      const response = await fetch(url, {credentials: 'same-origin'});
+      if (response.ok) {
+        return response.json();
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  },
+
+  async getIssue(issueId) {
+    if (!this.issueCache.has(issueId)) {
+      const issue = await this._get(`https://sentry.prod.mozaws.net/api/0/issues/${issueId}/`);
+      this.issueCache.set(issueId, issue);
+    }
+    return this.issueCache.get(issueId);
+  },
+
+  async getEvent(issueId, eventId='latest') {
+    const key = `${issueId}:${eventId}`;
+    if (!this.eventCache.has(key)) {
+      const url = (
+        `https://sentry.prod.mozaws.net/operations/nightly-js-errors/issues/` +
+        `${issueId}/events/${eventId}/json/`
+      );
+      const event = await this._get(url);
+      this.eventCache.set(key, event);
+    }
+    return this.eventCache.get(key);
+  },
+};
+
+const bugzilla = {
+  bugCache: new Map(),
+
+  async _get(url) {
+    try {
+      const response = await fetch(url, {headers: {Accept: 'application/json'}});
+      if (response.ok) {
+        return response.json();
+      }
+    } catch (err) {
+      console.error(err);
+    }
+  },
+
+  async getBugsForIssue(issueId) {
+      const url = new URL('https://bugzilla.mozilla.org/rest/bug');
+      url.searchParams.set('whiteboard', `[nightly-js-sentry:${issueId}]`);
+
+      if (!this.bugCache.has(issueId)) {
+        const data = await this._get(url.href);
```

Comment:
I'm worried about error handling here. If `_get` has a problem it returns undefined, which would then get cached, right? That doesn't seem good to me.

Reply:
This particular function is fine because it checks that `data` is truthy before storing it in the cache, but the event/issue fetching functions don't do the same, so I updated them to not cache `undefined`, and the bug URL function to check if they're defined before using them.